cvat/apps/engine/tests/test_rest_api.py

```diff
@@ -3385,6 +3399,15 @@ def _test_api_v2_tasks_id_data_spec(self, user, spec, data,
         response = send_data_callback(task_id, user, data)
         self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED, response.reason_phrase)
 
+        if get_status_callback:
+            response = get_status_callback(task_id, user)
+            while response.data['state'] not in ('Failed', 'Finished'):
+                response = get_status_callback(task_id, user)
```

Comment:
It's a potential infinite loop here, and it can waste  a lot of CPU here. I suggest adding a default timeout and some small delay between the requests (e.g. `sleep(0.01)`).